File: code-samples/webhooks/WebhookNotification.java
package WebHookNotification;

import java.io.IOException;
import com.google.common.reflect.TypeToken;
import com.google.gson.Gson;

import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class WebHookNotification {
    static RestClient restClient;
    static String DELIVERY_ADDRESS = "";
    public static void main(String[] args) {
      var obj = new WebHookNotification();
      try {
        // Instantiate the SDK
        restClient = new RestClient(System.getenv("RC_CLIENT_ID"), System.getenv("RC_CLIENT_SECRET"), System.getenv("RC_SERVER_URL"));

        // Authenticate a user using a personal JWT token
        restClient.authorize(System.getenv("RC_JWT"));

        // For the purpose of testing the code, we put the deliver address in the environment variable.
        // Feel free to set the delivery address directly.
        DELIVERY_ADDRESS = System.getenv("WEBHOOK_DELIVERY_ADDRESS") + "/webhook";

        obj.create_webhook_subscription();
        // obj.read_subscriptions;
      } catch (RestException e) {
        System.out.println(e.getMessage());
      } catch (IOException e) {
        e.printStackTrace();
      }
    }
    /*
    * Create a Webhok notification and subscribe for instant SMS message notification
    */
    private void create_webhook_subscription() throws RestException, IOException {
      try {
        var bodyParams = new CreateSubscriptionRequest();
        bodyParams.eventFilters = new String[] { "/restapi/v1.0/account/~/extension/~/message-store/instant?type=SMS" };
        bodyParams.deliveryMode = new NotificationDeliveryMode();
        bodyParams.deliveryMode.transportType = "WebHook";
        bodyParams.deliveryMode.address = DELIVERY_ADDRESS;

        bodyParams.expiresIn(3600L);

        var resp = restClient.restapi().subscription().post(bodyParams);
        System.out.println("Subscription Id: " + resp.id);
        System.out.println("Ready to receive incoming SMS via WebHook.");
      }catch(RestException e) {
        System.out.println(e.getMessage());
      }
    }
    /*
    * Read all created subscriptions
    */
    private void read_subscriptions()  throws RestException, IOException {
      try {
        var resp = restClient.restapi().subscription().list();
        if (resp.records.length == 0) {
          System.out.println("No subscription.");
        } else {
          for (var record : resp.records) {
            String jsonStr = new Gson().toJson(record, new TypeToken<Object>(){}.getType());
            System.out.println(jsonStr);
            delete_subscription(record.id);
          }
        }

      }catch(RestException e) {
        System.out.println(e.getMessage());
      }
    }
    /*
    * Delete a subscription identified by the subscription id
    */
    private void delete_subscription(String subscriptionId)  throws RestException, IOException {
      try {
        var resp = restClient.restapi().subscription(subscriptionId).delete();
        System.out.println("Subscription Id: " + subscriptionId + " deleted.");
      }catch(RestException e) {
        System.out.println(e.getMessage());
      }
    }
}

File: code-samples/webhooks/webhook-notification.cs
using System;
using System.Threading.Tasks;
using RingCentral;
using dotenv.net;
using System.Text.Json;


namespace setup_webhook
{
    class Program
    {
        static RestClient restClient;
        static string DELIVERY_ADDRESS = "";

        static async Task Main(string[] args)
        {
            DotEnv.Load();

            // For the purpose of testing the code, we put the deliver address in the environment variable.
            // Feel free to set the delivery address directly.
            DELIVERY_ADDRESS = Environment.GetEnvironmentVariable("WEBHOOK_DELIVERY_ADDRESS") + "/webhook";

            try
            {
                // Instantiate the SDK
                restClient = new RestClient(
                    Environment.GetEnvironmentVariable("RC_CLIENT_ID"),
                    Environment.GetEnvironmentVariable("RC_CLIENT_SECRET"),
                    Environment.GetEnvironmentVariable("RC_SERVER_URL"));


                // Authenticate a user using a personal JWT token
                await restClient.Authorize(Environment.GetEnvironmentVariable("RC_JWT"));
                await subscribe_for_notification();
                //  await read_subscriptions();
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
        }
        /*
        * Create a Webhok notification and subscribe for instant SMS message notification
        */
        static private async Task subscribe_for_notification()
        {
            try
            {
                var bodyParams = new CreateSubscriptionRequest();
                bodyParams.eventFilters = new[] { "/restapi/v1.0/account/~/extension/~/message-store/instant?type=SMS" };
                bodyParams.deliveryMode = new NotificationDeliveryModeRequest()
                {
                    transportType = "WebHook",
                    address = DELIVERY_ADDRESS
                };
                bodyParams.expiresIn = 3600;

                var resp = await restClient.Restapi().Subscription().Post(bodyParams);
                Console.WriteLine("Subscription Id: " + resp.id);
                Console.WriteLine("Ready to receive incoming SMS via WebHook.");
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
        }
        /*
        * Read all created subscriptions
        */
        static private async Task read_subscriptions()
        {
            try
            {
                var resp = await restClient.Restapi().Subscription().List();
                if (resp.records.Length == 0)
                {
                    Console.WriteLine("No subscription.");
                }
                else
                {
                    foreach (var record in resp.records)
                    {
                        Console.WriteLine(JsonSerializer.Serialize(record));
                        await delete_subscription(record.id);
                    }
                }

            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
        }
        /*
        * Delete a subscription identified by the subscription id
        */
        static private async Task delete_subscription(String subscriptionId)
        {
            try
            {
                var resp = await restClient.Restapi().Subscription(subscriptionId).Delete();
                Console.WriteLine("Subscription " + subscriptionId + " deleted.");
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
        }
    }
}
